### PR TITLE
adds validate_and_transform_graphql_field_name

### PR DESCRIPTION
**Pull request by bwalsh**
## Description

This PR improves the GraphQL field name transformation logic in validate_and_transform_graphql_field_name within gen3_tracker/meta/dataframer.py. The updated function now correctly handles leading spaces and special characters, ensuring all generated field names comply with GraphQL naming rules. This resolves test failures related to field name validation and transformation. No changes were made to the core FHIR database logic or resource flattening methods.

## Motivation and Context
Reported guppy issues #140 

## How Has This Been Tested?

**PR Description**

Adds comprehensive parameterized tests for the `validate_and_transform_graphql_field_name` function in `tests/unit/dataframer/test_dataframer.py`. The tests cover a wide range of input cases, including field names with spaces, special characters, reserved GraphQL introspection names, and edge cases like empty strings and numeric prefixes. This ensures robust validation and transformation of field names for GraphQL compatibility.

**Highlights:**
- Parameterized test covers multiple input/output scenarios.
- Validates handling of leading/trailing spaces, special characters, and reserved names.
- Enhanced field name sanitization for GraphQL compatibility.
- Fixed edge cases with leading spaces and reserved double underscores.

## Types of Changes
- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist
- [ ] I have updated the documentation accordingly (link here).  -- `NA - this fix is not user facing`
- [x ] I have tested that this feature locally.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [ ] Reviewer has tested this feature locally

### CODE REVIEW

File: gen3_tracker/common/__init__.py [hunk 1]
```diff
@@ -376,9 +376,9 @@ class Commit(BaseModel):
     """The exceptions."""
     logs: list = None
     """The logs."""
-    path: pathlib.Path = None
+    path: Optional[pathlib.Path] = None
```
Comment (bwalsh) [inline gen3_tracker/common/__init__.py:379]: @quinnwai Is there a test case for this?

Comment (quinnwai) [reply gen3_tracker/common/__init__.py:379]: Yes added one, it addresses a serialization error during publish where

> /Users/wongq/projects/gen3_util/venv/lib/python3.12/site-packages/pydantic/main.py:519: UserWarning: Pydantic serializer warnings:
> PydanticSerializationUnexpectedValue(Expected `<class 'pathlib.Path'>` but got `<class 'NoneType'>` with value `'None'` - serialized value may not be as expected.)
> PydanticSerializationUnexpectedValue(Expected `<class 'pathlib.Path'>` but got `<class 'NoneType'>` with value `'None'` - serialized value may not be as expected.)

It should just isolate the Commit object that, when `model_dump`ed, errors out if a Pathlib.path is assigned a None value.

**Comment by quinnwai** [comment]
Cool did a final test uploading invalid data. Worked for me.

**Review by quinnwai — APPROVED** [review]
